docs/applications/nublado/major-upgrade.rst

```diff
@@ -0,0 +1,313 @@
+   functionality of the latter two are subsumed in nublado v3).
+
+Enable nublado v3
+-----------------
+
+Pre-deployment Preparation
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The following tasks can be performed at any time, and should be
+performed ahead of your maintenance window to actually turn on the
+application.
+
+Vault Secret Creation
+"""""""""""""""""""""
+
+The first thing you will need to do is to set up secrets for
+``nublado`` (in future versions, this will require copying ``nublado``
+to ``nubladoN``, but ``nublado2`` is the current name for version 2, so
+if you're running an RSP, you already have those secrets set up
+correctly).
+
+These secrets, stored in Vault, and presented by
+``vault-secrets-operator`` are:
+
+#. ``cryptkeeper_key``: may be randomly generated; used internally by
+   the Hub to encrypt authentication state in the session database.
+#. ``crypto_key``: may be randomly generated; used to encrypt cookies
+   for user state in the browser session.
+#. ``proxy_token``: may be randomly generated; used to encrypt
+   communications between the JupyterHub Configurable HTTP Proxy and the
+   Hub.
+#. ``hub_db_password``: the password for the Hub session database; in
+   nublado v3 this is a Postgres database; conventionally both the
+   username and the database name are ``nublado3``.  In general you
+   should be consuming a database as a service, in which case you will
+   need to request this database from your DBAs and ask them the
+   password.  If you are instead running the internal database, you will
+   need to create this database and grant permissions in the
+   ``postgres`` application configuration.
+#. ``slack_webhook``: the Slack webhook allowing the JupyterLab
+   controller to post status messages to Slack.  This is the same one as
+   used by, e.g. Gafaelfawr in your RSP instance.
+
+Note that in future the secret setup and management will all be handled
+as part of Phalanx deployment, and you will no longer need to do this
+manually.  However, at the present, you still do.
+
+Enable the application
+""""""""""""""""""""""
+
+In the values file for your installation in the top-level Phalanx
+``environments`` directory, set ``nublado: true``.
+
+Configure nublado v3
+""""""""""""""""""""
+
+In the values file for your installation in ``applications/nublado``,
+set up your site-specific values.  The bulk of this work will be
+specifying the set of mount points you use for persistent user data.
+More than likely, you will use the same mount points for nublado v3 as
+you did for nublado v2, but the configuration syntax differs.
+
+Note that if you do share persistent storage between nublado versions,
+the step about communicating with your users (below) goes from
+"important" to "critical".
+
+
+Create an alternate route
+"""""""""""""""""""""""""
+
+In the values file for your installation in ``applications/nublado``,
+set the key ``jupyterhub.hub.baseUrl`` to ``/n3``.
```

Comment:
yeah this kind of feels like it should be applications/nublado3 (which I assume already exists so you can test it on the environment?) /n3 seems like the right thing and the thing you would change to /nb when you want to roll it out to everyone?

Reply:
See above.  As Cletus put it, "Shoulda, but didn't."